File: data/recomposed_projects/commons-fileupload/src/main/org/apache/commons/fileupload/FileUpload.py
# Imports Begin
from src.main.org.apache.commons.fileupload.FileUploadBase import *
from src.main.org.apache.commons.fileupload.FileItemFactory import *

# Imports End


class FileUpload(FileUploadBase):

    # Class Fields Begin
    __fileItemFactory: FileItemFactory = None
    # Class Fields End

    # Class Methods Begin
    def setFileItemFactory(self, factory: FileItemFactory) -> None:

        self.__fileItemFactory = factory

    def getFileItemFactory(self) -> FileItemFactory:

        return self.__fileItemFactory

    def __init__(self, constructorId: int, fileItemFactory: FileItemFactory) -> None:

        super().__init__()
        if constructorId == 1:
            self.__fileItemFactory = fileItemFactory

    # Class Methods End

File: java_projects/glue_code/commons-fileupload/src/main/java/org/apache/commons/fileupload/FileUpload.java
package org.apache.commons.fileupload;

import org.graalvm.polyglot.Value;

public class FileUpload extends FileUploadBase {
  private static Value clz = ContextInitializer.getPythonClass("FileUpload.py", "FileUpload");
  private Value obj;

  public FileUpload(Value obj) {
    this.obj = obj;
  }

  public Value getPythonObject() {
    return obj;
  }

  public void setFileItemFactory(FileItemFactory factory) {
    //
    // this.fileItemFactory = factory;
    //

    obj.invokeMember("setFileItemFactory", factory);
  }

  public FileItemFactory getFileItemFactory() {
    //
    // return fileItemFactory;
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("getFileItemFactory").as(FileItemFactory.class);
  }

  public FileUpload(int constructorId, FileItemFactory fileItemFactory) {
    super();
    // if (constructorId == 1) {
    // this.fileItemFactory = fileItemFactory;
    // }
    //

    this.obj = clz.invokeMember("__init__", constructorId, fileItemFactory);
  }
}
